src/main/java/com/example/bookstoreweb/service/UserService.java

```diff
@@ -3,10 +3,13 @@
 import com.example.bookstoreweb.dto.user.UserRegistrationRequestDto;
 import com.example.bookstoreweb.dto.user.UserResponseDto;
 import com.example.bookstoreweb.exception.RegistrationException;
+import com.example.bookstoreweb.model.User;
 
 public interface UserService {
     UserResponseDto register(UserRegistrationRequestDto registrationRequestDto)
             throws RegistrationException;
 
     UserResponseDto getByEmail(String email);
```

Comment:
I prefer services to return Model, not Dtos. And map model to Dtos in Controllers. So this way you wont need getByEmail + getUserByEmail

Reply:
But we need to use dto at the service level, and models at the repository level, to save information and make it available to users. Or not? Because I did everything by example in the video lectures

Reply:
That's just my vision. Not a problem to return DTO. The problem appears when you have getByEmail and getUserByEmail.  If you want your services to return DTOs. Then in your case where you had to put user into order, and save order. you'd better use userRepository.getByEmail(email), which return model, so this model can be put into order.